fabric-common-tags-api-v1/src/main/java/net/fabricmc/fabric/api/tags/v1/CommonItemTags.java

```diff
@@ -0,0 +1,75 @@
+package net.fabricmc.fabric.api.tags.v1;
+
+import net.minecraft.item.Item;
+import net.minecraft.tag.TagKey;
+
+import net.fabricmc.fabric.impl.v1.TagRegistration;
+
+public class CommonItemTags {//todo include fabric tools in common tools
+	public static final TagKey<Item> PICKAXES = register("pickaxes");
+	public static final TagKey<Item> SHOVELS = register("shovels");
+	public static final TagKey<Item> HOES = register("hoes");
+	public static final TagKey<Item> AXES = register("axes");
+	public static final TagKey<Item> SHEARS = register("shears");
+	/**
+	 * For throwable weapons, like Minecraft tridents
+	 */
+	public static final TagKey<Item> SPEARS = register("spears");
+	public static final TagKey<Item> SWORDS = register("swords");
+	public static final TagKey<Item> BOWS = register("bows");
+	/**
+	 * @deprecated Use {@link CommonItemTags#PICKAXES}
+	 */
+	@Deprecated
+	public static final TagKey<Item> FABRIC_PICKAXES = registerFabric("pickaxes");
+	/**
+	 * @deprecated Use {@link CommonItemTags#SHOVELS}
+	 */
+	@Deprecated
+	public static final TagKey<Item> FABRIC_SHOVELS = registerFabric("shovels");
+	/**
+	 * @deprecated Use {@link CommonItemTags#HOES}
+	 */
+	@Deprecated
+	public static final TagKey<Item> FABRIC_HOES = registerFabric("hoes");
+	/**
+	 * @deprecated Use {@link CommonItemTags#AXES}
+	 */
+	@Deprecated
+	public static final TagKey<Item> FABRIC_AXES = registerFabric("axes");
+	@Deprecated
+	public static final TagKey<Item> FABRIC_SHEARS = registerFabric("shears");
+	/**
+	 * @deprecated Use {@link CommonItemTags#SWORDS}
+	 */
+	@Deprecated
+	public static final TagKey<Item> FABRIC_SWORDS = registerFabric("swords");
+	public static final TagKey<Item> IRON_INGOTS = register("iron_ingots");
+	public static final TagKey<Item> IRON_ORES = register("iron_ores");
+	public static final TagKey<Item> GOLD_ORES = register("gold_ores");
+	public static final TagKey<Item> GOLD_INGOTS = register("gold_ingots");
+	public static final TagKey<Item> REDSTONE_DUSTS = register("redstone_dusts");
+	public static final TagKey<Item> REDSTONE_ORES = register("redstone_ores");
+	public static final TagKey<Item> COPPER_INGOTS = register("copper_ingots");
+	public static final TagKey<Item> COPPER_ORES = register("copper_ores");
+	public static final TagKey<Item> ORES = register("ores");
+	public static final TagKey<Item> NETHERITE_ORES = register("netherite_ores");
+	public static final TagKey<Item> NETHERITE_INGOTS = register("netherite_ingots");
+	public static final TagKey<Item> FOODS = register("foods");
+	public static final TagKey<Item> POTIONS = register("potions");
+	public static final TagKey<Item> BOOKSHELVES = register("bookshelves");
+
+	/**
+	 * Items in this tag are marked as cannot be placed into 'sub' or 'item' inventories, such as shulker boxes.
+	 * It is up to the inventory implementer to respect this tag's entries.
+	 */
+	public static final TagKey<Item> ITEM_INVENTORY_EXCLUDED = register("item_inventory_excluded");
```

Comment:
Is this functionality implemented for vanilla containers? I don't see any code here for it but I could just be missing it

Reply:
there isn't a tag functionality. Instead, this is controlled by a method on the item itself. This one (mojmap) 
https://github.com/TelepathicGrunt/Bumblezone-Fabric/blob/5ab009328eb13c1fdd1593cd950093fbe6b5d2ba/src/main/java/com/telepathicgrunt/the_bumblezone/items/BzBlockItem.java#L19

Overriding that will make your item unable to be placed in shulker boxes. Other container's slots should be checking that method to know if it can accept the item or not. This tag may not be needed tbh.

Reply:
Indeed if there's already a built-in method that should absolutely be used

My question was more about implementing this for shulker inventories though, as if this gets added it seems odd to not make it add that functionality to the one place the documentation calls out by name, at least

Reply:
This was a suggestion in the discord, and something @Technici4n was going to look at for transfer-api.

Reply:
Well I'm not sure there's much worth discussing here, there is already a vanilla method that implements this functionality, making this redundant at best and confusing for modders at worst

Reply:
The name is terrible
like, what is the inventory talked about here??

If I had to take it literally, then for me it means that anything in that tag would not be allowed in any inventories.

Reply:
Such a tag would probably have to be in the fabric namespace since that implements the feature and also offered by the module implementing it, not here.

Reply:
removed